### ISSUE TITLE
Using gb with golang.org/x/sys/unix fails

### ISSUE THREAD

**Issue by pnasrat** [constabulary/gb#82]
Steps to reproduce

* Install `gb` and `gb-vendor`
* `mkdir -p delve/src/github.com/derekparker/delve`
* `git clone https://github.com/derekparker/delve.git delve/src/github.com/derekparker/delve`
* `cd delve`
* `gb vendor golang.org/x/sys/unix`
* `gb vendor github.com/emicklei/go-restful`
* `gb vendor github.com/peterh/liner`
* `gb build -v all`

Results

OS X
```
# /usr/local/go/pkg/tool/darwin_amd64/6a -o /var/folders/ng/qv4q6r9n7sbcv8s5mjxj_yzm0000gn/T/gb028435063/golang.org/x/sys/unix/as.6 -D GOOS_darwin -D GOARCH_amd64 /Users/pnasrat/Development/delve/vendor/src/golang.org/x/sys/unix/asm_darwin_amd64.s
/Users/pnasrat/Development/delve/vendor/src/golang.org/x/sys/unix/asm_darwin_amd64.s:8 6a: No such file or directory: textflag.h
# /usr/local/go/pkg/tool/darwin_amd64/6a -o /var/folders/ng/qv4q6r9n7sbcv8s5mjxj_yzm0000gn/T/gb028435063/golang.org/x/sys/unix/as.6 -D GOOS_darwin -D GOARCH_amd64 /Users/pnasrat/Development/delve/vendor/src/golang.org/x/sys/unix/asm.s
/Users/pnasrat/Development/delve/vendor/src/golang.org/x/sys/unix/asm.s:8 6a: No such file or directory: textflag.h
```
Linux:
```
# /usr/local/go/pkg/tool/linux_amd64/6a -o /tmp/gb151227897/golang.org/x/sys/unix/as.6 -D GOOS_linux -D GOARCH_amd64 /home/pnasrat/delve/vendor/src/golang.org/x/sys/unix/asm.s
/home/pnasrat/delve/vendor/src/golang.org/x/sys/unix/asm.s:8 6a: No such file or directory: textflag.h
# /usr/local/go/pkg/tool/linux_amd64/6a -o /tmp/gb151227897/golang.org/x/sys/unix/as.6 -D GOOS_linux -D GOARCH_amd64 /home/pnasrat/delve/vendor/src/golang.org/x/sys/unix/asm_linux_amd64.s
/home/pnasrat/delve/vendor/src/golang.org/x/sys/unix/asm_linux_amd64.s:8 6a: No such file or directory: textflag.h
DEBUG build duration: 720.273975ms map[compile:688.508329ms asm:46.884575ms]
FATAL command "build" failed: exit status 1
```

go tool commands for `go get -u -v -x golang.org/x/sys/unix`

```
/usr/local/go/pkg/tool/linux_amd64/6g -o $WORK/golang.org/x/sys/unix.a -trimpath $WORK -p golang.org/x/sys/unix -D _/home/pnasrat/go/src/golang.org/x/sys/unix -I $WORK -pack ./constants.go ./env_unix.go ./env_unset.go ./flock.go ./race0.go ./sockcmsg_linux.go ./sockcmsg_unix.go ./str.go ./syscall.go ./syscall_linux.go ./syscall_linux_amd64.go ./syscall_unix.go ./zerrors_linux_amd64.go ./zsyscall_linux_amd64.go ./zsysnum_linux_amd64.go ./ztypes_linux_amd64.go
/usr/local/go/pkg/tool/linux_amd64/6a -trimpath $WORK -I $WORK/golang.org/x/sys/unix/_obj/ -I /usr/local/go/pkg/linux_amd64 -o $WORK/golang.org/x/sys/unix/_obj/asm.6 -D GOOS_linux -D GOARCH_amd64 ./asm.s
/usr/local/go/pkg/tool/linux_amd64/6a -trimpath $WORK -I $WORK/golang.org/x/sys/unix/_obj/ -I /usr/local/go/pkg/linux_amd64 -o $WORK/golang.org/x/sys/unix/_obj/asm_linux_amd64.6 -D GOOS_linux -D GOARCH_amd64 ./asm_linux_amd64.s
```

**Comment by pnasrat** [comment]
This may be a dupe of  issue #12

**Comment by davecheney** [comment]
I don't think it is #12, there is no cgo used in that repo that I know of
(I could be wrong)

Thanks for the repo, I'll check it out.

**Comment by davecheney** [comment]
Thanks for the report. I think I've fixed this specific bug, but there may be others, please test it out and let me know.

**Comment by enr** [comment]
On Linux and following @pnasrat 's intructions I still have errors (differents from errors using the previous gb version):

```
vendor/src/golang.org/x/sys/unix/syscall_linux_arm64.go:65: Getpagesize redeclared in this block
	previous declaration at vendor/src/golang.org/x/sys/unix/syscall_linux_amd64.go:73
vendor/src/golang.org/x/sys/unix/syscall_linux_arm64.go:70: TimespecToNsec redeclared in this block
	previous declaration at vendor/src/golang.org/x/sys/unix/syscall_linux_amd64.go:87
vendor/src/golang.org/x/sys/unix/syscall_linux_arm64.go:72: NsecToTimespec redeclared in this block
	previous declaration at vendor/src/golang.org/x/sys/unix/syscall_linux_amd64.go:89
[...]
FATAL command "build" failed: exit status 1

```

**Comment by davecheney** [comment]
This might not be gb's fault, go 1.4 doesn't know that arm64 is a $GOARCH,
so it's messing up the build constraint.

**Comment by davecheney** [comment]
I think this is a bug upstream, the /x/sys repo doesn't build using go 1.4 any more
```
lucky(~/devel/demo8/delve) % env GOPATH=$(pwd):$(pwd)/vendor $HOME/go1.4/bin/go build .../x/...
# golang.org/x/sys/unix
vendor/src/golang.org/x/sys/unix/syscall_linux_arm64.go:65: Getpagesize redeclared in this block
        previous declaration at vendor/src/golang.org/x/sys/unix/syscall_linux_amd64.go:73
vendor/src/golang.org/x/sys/unix/syscall_linux_arm64.go:70: TimespecToNsec redeclared in this block
        previous declaration at vendor/src/golang.org/x/sys/unix/syscall_linux_amd64.go:87
vendor/src/golang.org/x/sys/unix/syscall_linux_arm64.go:72: NsecToTimespec redeclared in this block
        previous declaration at vendor/src/golang.org/x/sys/unix/syscall_linux_amd64.go:89
vendor/src/golang.org/x/sys/unix/syscall_linux_arm64.go:78: TimevalToNsec redeclared in this block
        previous declaration at vendor/src/golang.org/x/sys/unix/syscall_linux_amd64.go:95
vendor/src/golang.org/x/sys/unix/syscall_linux_arm64.go:80: NsecToTimeval redeclared in this block
        previous declaration at vendor/src/golang.org/x/sys/unix/syscall_linux_amd64.go:97
vendor/src/golang.org/x/sys/unix/syscall_linux_arm64.go:87: Pipe redeclared in this block
        previous declaration at vendor/src/golang.org/x/sys/unix/syscall_linux_amd64.go:106
vendor/src/golang.org/x/sys/unix/syscall_linux_arm64.go:100: Pipe2 redeclared in this block
        previous declaration at vendor/src/golang.org/x/sys/unix/syscall_linux_amd64.go:119
vendor/src/golang.org/x/sys/unix/syscall_linux_arm64.go:111: (*PtraceRegs).PC redeclared in this block
        previous declaration at vendor/src/golang.org/x/sys/unix/syscall_linux_amd64.go:130
vendor/src/golang.org/x/sys/unix/syscall_linux_arm64.go:113: (*PtraceRegs).SetPC redeclared in this block
        previous declaration at vendor/src/golang.org/x/sys/unix/syscall_linux_amd64.go:132
vendor/src/golang.org/x/sys/unix/syscall_linux_arm64.go:115: (*Iovec).SetLen redeclared in this block
        previous declaration at vendor/src/golang.org/x/sys/unix/syscall_linux_amd64.go:134
vendor/src/golang.org/x/sys/unix/syscall_linux_arm64.go:115: too many errors
```
I'll see if I can get that fixed upstream

**Comment by enr** [comment]
You are right, I have the same error using the canonical `go get`.

Thank you for the effort in this tool.

Enrico

**Comment by davecheney** [comment]
See golang/go#10854

**Comment by davecheney** [comment]
Upstream have fixed golang/go#10854. I have confirmed this package builds with 1.4.x, which is the Go release gb expects.
